superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -117,30 +116,20 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
     );
   }
 
-  componentDidMount() {
-    this.props.actions.fetchDashboards(this.props.userId).then(() => {
-      if (ensureIsArray(this.props.dashboards).length === 0) {
-        return;
-      }
-      const dashboardIds = this.props.dashboards?.map(
-        dashboard => dashboard.value,
-      );
+  async componentDidMount() {
+    let { dashboardId } = this.props;
+    if (!dashboardId) {
       const lastDashboard = sessionStorage.getItem(SK_DASHBOARD_ID);
-      let recentDashboard = lastDashboard && parseInt(lastDashboard, 10);
-
-      if (this.props.dashboardId) {
-        recentDashboard = this.props.dashboardId;
-      }
-
-      if (
-        recentDashboard !== null &&
-        dashboardIds.indexOf(recentDashboard) !== -1
-      ) {
+      dashboardId = lastDashboard && parseInt(lastDashboard, 10);
+    }
+    if (dashboardId) {
+      const { result } = await this.loadDashboard(dashboardId);
```

Comment:
It should be better to add the error message.

```suggestion
   try {
     const { result } = await this.loadDashboard(dashboardId);
     if (result) {
         ...
     }
     catch (e) {
       this.props.addDangerToast(t('An error occurred while loading dashboard information.'));
     }
```

Reply:
Done